emcc.py

```diff
@@ -1881,7 +1887,7 @@ def get_clang_command_asm(src_file):
         return 0
 
       # Precompiled headers support
-      if has_header_inputs:
+      if has_header_inputs or 'header' in language_mode:
```

Comment:
is this change unrelated? i can't connect it.

Reply:
This technically a more correct behaviour so we definitely want to land it.  I'm trying to remember why it is needed for this change .. it cause a test to fail without it.  I can check to confirm which one now.

Reply:
Tried reverting this.  Will land it later as part of a separate change (if this passes)